File: backend/backend/src/SignalingComponent/Common/DTOs/CloseConnectionMessage.cs
using System.Text.Json.Serialization;
using backend.WebSocketComponent.Common.DTOs;

namespace backend.SignalingComponent.Common.DTOs;

public class CloseConnectionMessage : ITypedMessage
{
    public static string TypeString => "close-connection";

    [JsonIgnore]
    public string InstanceTypeString => TypeString;

    [JsonPropertyName("requestID")]
    public string? RequestId { get; set; }
    
    [JsonPropertyName("remoteToken")]
    public required string RemoteToken { get; set; }
}

File: frontend/src/types/connection/CloseConnectionMessage.ts
import { ITypedMessage } from "../ITypedMessage";
import { MessageType } from "../MessageType";

export class CloseConnectionMessage implements ITypedMessage {
    public readonly type = MessageType.CLOSE_CONNECTION;
    public msg: {
        requestID?: string;
        remoteToken: string;
    };

    public constructor(msg: { requestID?: string; remoteToken: string }) {
        this.msg = msg;
    }
}
